Automattic/jetpack
# cli: Add dependency analysis helper and command

#### Changes proposed in this Pull Request:

This is the existing logic from tools/find-project-deps.php,
tools/get-build-order.php, and .github/files/list-changed-projects.php,
rewritten in JS. It also has a little more flexibility as far as
processing only a subset of projects.

One difference is in the unused composer.json `.extra.dependencies`
array: it will want that to instead be an object holding
`.extra.dependencies.test` and `.extra.dependencies.build`.

Changes to make use of this in place of tools/find-project-deps.php,
tools/get-build-order.php, and .github/files/list-changed-projects.php
will be done as a followup.

#### Jetpack product discussion


p9dueE-4dJ-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Try the new command, see how it works. Compare its output to the existing scripts.
